front_end/content.py

```diff
@@ -248,6 +261,95 @@ def get_problem_details(course, assignment, problem, format_content=False, parse
 
     return problem_dict
 
+def split_url(file_path):
+
+    id_dict = {}
+    id_dict["name"] = ""
+    id_dict["course_id"] = ""
+    id_dict["assignment_id"] = ""
+    id_dict["problem_id"] = ""
+    re_match = re.match(r"^\/(?P<name>[^\/]*)\/(?P<course_id>[^\/]*)\/?(?P<assignment_id>[^\/]*)\/?(?P<problem_id>[^\/]*)", file_path)
+    if re_match:
+        id_dict = re_match.groupdict()
+        for key, value in id_dict.items():
+            if value == None:
+                id_dict[key] = ""
+    return id_dict
+
+def get_titles(file_path):
```

Comment:
Please make this function name more descriptive. The function name is very generic, but it seems that it is accomplishing a specific purpose. Also consider whether content.py is the best place for this function.  If this function is specific to log summarization, it should go in that script. But if it can be used more generally, it can go here.